### PR TITLE
Add solid lines on Android canvas

### PR DESCRIPTION
**Pull request by paulproteus**
Add drawing solid lines on Android canvas.

This makes progress toward implementing the Canvas widget on Android generally.

## PR Checklist:
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

## Testing done

I made a sample app here: https://github.com/paulproteus/toganvas/blob/main/toganvas/src/toganvas/app.py / https://github.com/paulproteus/toganvas

On iOS and on Android, it looks as follows.

image

image

There's lots more work to do, but I thought it would be nice for my motivation to submit some progress via this pull request.

### CODE REVIEW

File: src/android/toga_android/widgets/canvas.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+from ..libs import activity
+from ..libs.android.graphics import Paint, Paint__Style, Path
+from .base import Widget
+
+# Arbitrary scale factor; to be made more specific in the future.
+SCALE_FACTOR = 5
```
Comment (freakboy3742) [inline src/android/toga_android/widgets/canvas.py:6]: This stood out for me; I'm guessing this is a factor that should be tied to a combination of screen DPI and size? As it stands, it looks like the Android canvas draws "bigger" than the iOS equivalent

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
There's obviously a lot to come before there's a full canvas - but the drawing pieces here look pretty straightforward. There are only 2 pieces that stand out for me:

1) Scale factor; you've already flagged this as a TODO of sorts; is the intention to get this working now as part of this PR, or later?
2) I'm a little hazy about the relationship between the native widgets here. I know we added the View and IView to the base template so that we've got something concrete to subclass, but this code then makes it looks like it's chasing it's own tail - we define an instance of CustomView; then define a Python view and set it on the view? It feels like there's one too many things called View in this setup... 

At the very least, some sort of inline documentation is required to clarify exactly what is going on here - if only so that the next ~~victim~~contributor doesn't have to scratch their head so much :-)

**Comment by paulproteus** [comment]
> Scale factor; you've already flagged this as a TODO of sorts; is the intention to get this working now as part of this PR, or later?

I hope to handle that soon, but outside this pull request if possible.

> I'm a little hazy about the relationship between the native widgets here. I know we added the View and IView to the base template so that we've got something concrete to subclass, but this code then makes it looks like it's chasing it's own tail - we define an instance of CustomView; then define a Python view and set it on the view? It feels like there's one too many things called View in this setup...

Good call!

https://github.com/beeware/briefcase-android-gradle-template/pull/32 accomplishes a rename.

I've updated this pull request to be based on that version of the template.

What do you think now?

File: src/android/toga_android/widgets/canvas.py (hunk 1 above)
Comment (paulproteus) [reply src/android/toga_android/widgets/canvas.py:6]: Definitely. I hope to fix this is in a follow-up.

**Review by freakboy3742 — APPROVED** [review]
👍

**Pull request merged**